tests/evaluation/test_elo_evaluator.py

```diff
@@ -0,0 +1,193 @@
+from typing import Sequence, Tuple
+
+from dotenv import load_dotenv
+from pytest import fixture
+
+from intelligence_layer.connectors import AlephAlphaClientProtocol
+from intelligence_layer.core import (
+    ControlModel,
+    Language,
+    LuminousControlModel,
+    TextChunk,
+    utc_now,
+)
+from intelligence_layer.core.tracer.tracer import NoOpTracer, Tracer
+from intelligence_layer.evaluation import (
+    ComparisonEvaluation,
+    EloEvaluationLogic,
+    EvaluationLogic,
+    Evaluator,
+    Example,
+    ExampleOutput,
+    InMemoryDatasetRepository,
+    InMemoryEvaluationRepository,
+    InMemoryRunRepository,
+    Matches,
+    MatchOutcome,
+    RunOverview,
+    SuccessfulExampleOutput,
+)
+from intelligence_layer.examples import SingleChunkQaInput, SingleChunkQaOutput
+
+load_dotenv()
+
+
+class DummyEloQaEvalLogic(
+    EloEvaluationLogic[SingleChunkQaInput, SingleChunkQaOutput, SingleChunkQaOutput]
+):
+    def __init__(
+        self,
+        model: ControlModel,
+        tracer: Tracer = NoOpTracer(),
+    ):
+        super().__init__()
+        self._model = model
+        self.tracer = tracer
+
+    def grade(
+        self,
+        first: SuccessfulExampleOutput[SingleChunkQaOutput],
+        second: SuccessfulExampleOutput[SingleChunkQaOutput],
+        example: Example[SingleChunkQaInput, SingleChunkQaOutput],
+    ) -> MatchOutcome:
+        _ = example
+        if first.run_id < second.run_id:
+            return MatchOutcome.A_WINS
+        elif first.run_id > second.run_id:
+            return MatchOutcome.B_WINS
+        else:
+            return MatchOutcome.DRAW
+
+
+@fixture
+def model(client: AlephAlphaClientProtocol) -> ControlModel:
+    return LuminousControlModel(client=client, name="luminous-base-control")
+
+
+@fixture
+def in_memory_dataset_repository() -> InMemoryDatasetRepository:
+    return InMemoryDatasetRepository()
+
+
+@fixture
+def in_memory_run_repository() -> InMemoryRunRepository:
+    return InMemoryRunRepository()
+
+
+@fixture
+def in_memory_evaluation_repository() -> InMemoryEvaluationRepository:
+    return InMemoryEvaluationRepository()
```

Comment:
I'm pretty sure these fixtures exist already in the top-level conftest

Reply:
Removed them :)